File: videosdk-live/videosdk-rtc-android-java-call-trigger-example/call-trigger-app/app/src/main/java/live/videosdk/call_trigger/example/Network/NetworkCallHandler.java
package live.videosdk.call_trigger.example.Network;

import android.util.Log;

import androidx.annotation.NonNull;

import java.util.HashMap;
import java.util.Map;

import live.videosdk.call_trigger.example.MainActivity;
import live.videosdk.call_trigger.example.MainApplication;
import live.videosdk.call_trigger.example.Services.MyFirebaseMessagingService;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public  class NetworkCallHandler {

    static ApiService apiService = ApiClient.getClient().create(ApiService.class);
    public static String FcmToken;
    public static String calleeInfoToken;

    public static void updateCall(String call_update) {

        String fcmToken = MyFirebaseMessagingService.FCMtoken;

        Map<String, String> callerInfo = new HashMap<>();
        Map<String, Object> callUpdateBody = new HashMap<>();

        //callerInfo
        callerInfo.put("callerId", MainActivity.myCallId);
        callerInfo.put("token",fcmToken);

        //CallUpdateBody
        callUpdateBody.put("callerInfo",callerInfo);
        callUpdateBody.put("type",call_update);

        Call<String> call = apiService.updateCall(callUpdateBody);
        call.enqueue(new Callback<String>() {
            @Override
            public void onResponse(@NonNull Call<String> call, @NonNull Response<String> response) {
                if (response.isSuccessful()) {
                    Log.d("API", "Call updated successfully: " + response.body());
                }
            }

            @Override
            public void onFailure(@NonNull Call<String> call, @NonNull Throwable t) {
                Log.e("API", "Call update failed", t);
            }
        });
    }


    public void initiateCall() {
        ApiService apiService = ApiClient.getClient().create(ApiService.class);

        Map<String,String> callerInfo = new HashMap<>();
        Map<String,String> calleeInfo = new HashMap<>();
        Map <String,String> videoSDKInfo= new HashMap<>();

        //callerInfo
        callerInfo.put("callerId",MainActivity.myCallId);
        callerInfo.put("token",FcmToken);

        //calleeInfo
        calleeInfo.put("token",calleeInfoToken);

        //videoSDKInfo
        videoSDKInfo.put("meetingId", MainApplication.getMeetingId());
        videoSDKInfo.put("token",MainApplication.getToken());

        Map<String,Object> callRequestBody = new HashMap<>();
        callRequestBody.put("callerInfo",callerInfo);
        callRequestBody.put("calleeInfo",calleeInfo);
        callRequestBody.put("videoSDKInfo",videoSDKInfo);

        Call<String> call = apiService.initiateCall(callRequestBody);
        call.enqueue(new Callback<String>() {
            @Override
            public void onResponse(@NonNull Call<String> call, @NonNull Response<String> response) {
                if (response.isSuccessful()) {
                    Log.d("API", "Call initiated: " + response.body());
                } else {
                    Log.e("API", "Failed to initiate call: " + response.message());
                }
            }

            @Override
            public void onFailure(@NonNull Call<String> call, @NonNull Throwable t) {
                Log.e("API", "API call failed: " + t.getMessage());
            }
        });
    }
}

File: videosdk-live/videosdk-rtc-android-kotlin-call-trigger-example/call-trigger-app/app/src/main/java/live/videosdk/call_trigger/kotlin/example/Network/NetworkCallHandler.kt
package live.videosdk.call_trigger.kotlin.example.Network

import android.util.Log
import live.videosdk.call_trigger.kotlin.example.MainApplication

import live.videosdk.call_trigger.kotlin.example.Services.MyFirebaseMessagingService
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class NetworkCallHandler  {

    companion object{
        lateinit var myCallId: String
        lateinit var FcmToken: String
        lateinit var calleeInfoToken:String
    }
        fun updateCall(call_update: String){
        val fcmToken: String = MyFirebaseMessagingService.FCMtoken

        val callerInfo: MutableMap<String, String> = HashMap()
        val callUpdateBody: MutableMap<String, Any> = HashMap()

        //callerInfo
        callerInfo["callerId"] = myCallId
        callerInfo["token"] = fcmToken
        //CallUpdateBody
        callUpdateBody["callerInfo"] = callerInfo
        callUpdateBody["type"] = call_update

        val apiService = ApiClient.client!!.create(ApiService::class.java)
        val call : Call<String> = apiService.updateCall(callUpdateBody)
        call.enqueue(object :Callback<String>{
            override fun onFailure(call: Call<String>, t: Throwable) {
                Log.d("TAG", "onFailure: "+ t.message)
            }

            override fun onResponse(call: Call<String>, response: Response<String>) {
                Log.d("TAG", "Call updated successfully: " + response.body())
            }
        })
    }


    fun initiateCall() {
        val apiService: ApiService = ApiClient.client!!.create(ApiService::class.java)

        val callerInfo: MutableMap<String, String> = HashMap()
        val calleeInfo: MutableMap<String, String> = HashMap()
        val videoSDKInfo: MutableMap<String, String> = HashMap()

        // callerInfo
        callerInfo["callerId"] = myCallId
        callerInfo["token"] = FcmToken
        // calleeInfo
        calleeInfo["token"] = calleeInfoToken
        // videoSDKInfo
        videoSDKInfo["meetingId"] = MainApplication.meetingId ?: return
        videoSDKInfo["token"] = MainApplication.token

        val callRequestBody: MutableMap<String, Any> = HashMap()
        callRequestBody["callerInfo"] = callerInfo
        callRequestBody["calleeInfo"] = calleeInfo
        callRequestBody["videoSDKInfo"] = videoSDKInfo

        val call: Call<String> = apiService.initiateCall(callRequestBody)
        call.enqueue(object : Callback<String> {
            override fun onResponse(call: Call<String>, response: Response<String>) {
                if (response.isSuccessful) {
                    Log.d("API", "Call initiated: " + response.body())
                } else {
                    Log.e("API", "Failed to initiate call: " + response.message())
                }
            }

            override fun onFailure(call: Call<String>, t: Throwable) {
                Log.e("API", "API call failed: " + t.message)
            }
        })
    }
}